pkg/dao/metrics.go

```diff
@@ -104,3 +105,40 @@ func (d metricsDaoImpl) PublicRepositoriesFailedIntrospectionCount(ctx context.C
 		Count(&output)
 	return int(output)
 }
+
+func (d metricsDaoImpl) PendingTasksCount(ctx context.Context) int64 {
+	var output int64 = -1
+	res := d.db.WithContext(ctx).
+		Model(&models.TaskInfo{}).
+		Where("status = ?", config.TaskStatusPending).Count(&output)
+	if res.Error != nil {
+		return output
+	}
+	return output
+}
+
+func (d metricsDaoImpl) PendingTasksAverageLatency(ctx context.Context) float64 {
+	var output float64
+	res := d.db.WithContext(ctx).
+		Raw("SELECT  extract(epoch from (COALESCE(AVG(AGE(CURRENT_TIMESTAMP, queued_at)), INTERVAL '0 days'))) as latency FROM tasks WHERE status = ?", config.TaskStatusPending).
+		Pluck("latency", &output)
+	if res.Error != nil {
+		return output
+	}
+
+	return output
+}
+
+func (d metricsDaoImpl) PendingTasksOldestTask(ctx context.Context) float64 {
+	var task models.TaskInfo
+	res := d.db.WithContext(ctx).
+		Model(&models.TaskInfo{}).
+		Where("status = ?", config.TaskStatusPending).Order("queued_at ASC").First(&task)
```

Comment:
Would we want to alert on this one? IIRC, when a failed task is requeued, the queued_at doesn't reset, so this might cause some false positives. It might make sense to reset the queued_at time.